framework/src/actions/CreateExecutionerAction.C

```diff
@@ -151,15 +154,94 @@ CreateExecutionerAction::storeCommonExecutionerParams(FEProblem & fe_problem, In
+     */
+    if (*it == "-log_summary")
+      mooseError("The PETSc option \"-log_summary\" can only be used on the command line.  Please remove it from the input file");
+
+    // Warn about superseded PETSc options (Note: -snes is not a REAL option, but people used it in their input files)
+    else
+    {
+      std::string help_string;
+      if (*it == "-snes" || *it == "-snes_mf" || *it == "-snes_mf_operator")
+        help_string = "Please set the solver type through \"solve_type\".";
+      else if (*it == "-ksp_monitor")
+        help_string = "Please use \"Outputs/print_linear_residuals=true\"";
+
+      if (help_string != "")
+        mooseWarning("The PETSc option " << *it << " should not be used directly in a MOOSE input file. " << help_string);
+    }
+
+    // Update the stored items, but do not create duplicates
+    if (find(po.flags.begin(), po.flags.end(), *it) == po.flags.end())
+      po.flags.push_back(*it);
+  }
+
+  // Check that the name value pairs are sized correctly
+  if (petsc_options_inames.size() != petsc_options_values.size())
+    mooseError("PETSc names and options are not the same length");
+
+  // Setup the name value pairs
+  bool boomeramg_found = false;
+  bool strong_threshold_found = false;
+  std::string pc_description = "";
+  for (unsigned int i = 0; i < petsc_options_inames.size(); i++)
+  {
+    // Do not add duplicate settings
+    if (find(po.inames.begin(), po.inames.end(), petsc_options_inames[i]) == po.inames.end())
+    {
+      po.inames.push_back(petsc_options_inames[i]);
+      po.values.push_back(petsc_options_values[i]);
+
+      // Look for a pc description
+      if (petsc_options_inames[i] == "-pc_type" || petsc_options_inames[i] == "-pc_sub_type" || petsc_options_inames[i] == "-pc_hypre_type")
+        pc_description += petsc_options_values[i] + ' ';
+
+      // This special case is common enough that we'd like to handle it for the user.
+      if (petsc_options_inames[i] == "-pc_hypre_type" && petsc_options_values[i] == "boomeramg")
+        boomeramg_found = true;
+      if (petsc_options_inames[i] == "-pc_hypre_boomeramg_strong_threshold")
+        strong_threshold_found = true;
+    }
+    else
+    {
+      for (unsigned int j = 0; j < po.inames.size(); j++)
+        if (po.inames[j] == petsc_options_inames[i])
+          po.values[j] = petsc_options_values[i];
+    }
+  }
+
+
+  // When running a 3D mesh with boomeramg, it is almost always best to supply a strong threshold value
+  // We will provide that for the user here if they haven't supplied it themselves.
+  if (boomeramg_found && !strong_threshold_found && fe_problem.mesh().dimension() == 3)
+  {
+    po.inames.push_back("-pc_hypre_boomeramg_strong_threshold");
+    po.values.push_back("0.7");
+    pc_description += "strong_threshold: 0.7 (auto)";
+  }
+
+  // Set Preconditioner description
+  fe_problem.setPreconditionerDescription(pc_description);
```

Comment:
All of this logic is duplicated in `FEProblem::storePetscOptions()` I don't see it being removed in this diff.